myapp/app/controllers/tasks_controller.rb

```diff
@@ -1,21 +1,22 @@
 class TasksController < ApplicationController
   before_action :set_task, only: [:show, :edit, :update, :destroy]
   before_action :get_users, only: [:new, :edit]
+  before_action :check_current_user, only: [:edit, :destroy]
 
   def index
-    @tasks = Task.eager_load(:user).all.page(params[:page])
-
+    @tasks = current_user.tasks.eager_load(:user).all.page(params[:page])
   end
 
   def show
+    current_user.tasks.find(params[:id])
```

Comment:
ちょっと自信ないですが@taskいれないとだめかな？
詳細画面で正しく値が入るかどうか確認お願いします。
```suggestion
    @task = current_user.tasks.find(params[:id])
```

Reply:
おかしな構造になってしまっていたため、修正しました。
修正前：
before_actionでタスク取得メソッドを呼び、`current_user.tasks.find(params[:id])`でログインユーザのタスクかチェックする
修正後：
`before_action`で`set_task`メソッド（ログインユーザのタスクを取得して@taskに代入）を呼ぶ